### ISSUE TITLE
Async chunks not being collected by the webExtractor.collectChunks() call for SSR of async routes with split chunks

### ISSUE THREAD

**Issue by cbdeveloper** [gregberge/loadable-components#722]
I working on this test repo to implement SSR + Code Splitting with `@loadable/component`.

https://github.com/cbdeveloper/test-loadable-components

It's currently working, but I don't think I'm getting the proper script tags for each route. I'm always getting the same script tags and the one that specific to the route is not being added.

These are my routes:

```
import React from "react";
import { ROUTE } from "./ROUTE";
import { Route, Switch } from "react-router-dom";
import Home from "../pages/Home";
import loadable from "@loadable/component";

interface AllRoutes {}

const AsyncPageA = loadable(() => import(/* webpackChunkName: "page-a" */ "../pages/PageA"));
const AsyncPageB = loadable(() => import(/* webpackChunkName: "page-b" */ "../pages/PageB"));

export const AllRoutes: React.FC<AllRoutes> = () => {
  return (
    <Switch>
      <Route exact path={ROUTE.HOME} component={Home}/>
      <Route exact path={ROUTE.PAGE_A} component={AsyncPageA}/>
      <Route exact path={ROUTE.PAGE_B} component={AsyncPageB}/>
    </Switch>
  );
};
```

So I'm splitting the chunks for Page A and Page B.

These are the generated thunks (for both node and web). They seem to make sense:

That hash in the file name is a `contenthash`.

![image](https://user-images.githubusercontent.com/43407798/111306548-3d5ddf80-8650-11eb-9543-d1ba1a80d32e.png)

This is my `ssrApp` handler:

```
export const ssrApp: RequestHandler = (req,res,next) => {
  const nodeStats = path.resolve(__dirname,"../../dist/node/loadable-stats.json");
  const webStats  = path.resolve(__dirname,"../../dist/web/loadable-stats.json");

  const nodeExtractor = new ChunkExtractor({ statsFile: nodeStats, entrypoints: ["main"] });
  const webExtractor = new ChunkExtractor({ statsFile: webStats, entrypoints: ["main"] })

  const { default: App } = nodeExtractor.requireEntrypoint();

  const jsx = webExtractor.collectChunks(
    <StaticRouter location={req.path} context={{}}>
      <App/>
    </StaticRouter>
  );

  const scriptTags = webExtractor.getScriptTags();

  console.log(`=== SCRIPT TAGS ===`);
  console.log(scriptTags);
  console.log(`===================`);

  const html = renderToString(jsx);
  const fullHtml = generateHtml(html,scriptTags);

  return res.send(fullHtml);
};
```

**What I was expecting?**

I was expecting that for the async routes like `/page-a` and `page-b` the `webExtractor.getScriptTags()` call would add the `page-a` or `page-b` async chunk to the HTML. And that is not the case.

This is what I get when I hit the home page (which does not load any of the split chunks) `/`

![image](https://user-images.githubusercontent.com/43407798/111307591-9417e900-8651-11eb-896c-078a6db3de9a.png)

Only the initial chunks are added. This is expected, because the Home route is not split.

But when I hit `/page-a`, I get the very same set of scripts:

![image](https://user-images.githubusercontent.com/43407798/111307678-b0b42100-8651-11eb-99e8-92eac3b43fca.png)

And I was expecting to also get the `page-a.43b04.js` script for the split chunk.

Am I doing something wrong or this is exactly how it's supposed to work?

I'm guessing that if I'm collecting chunks for that specific render:

```
const jsx = webExtractor.collectChunks(
    <StaticRouter location={req.path} context={{}}>
      <App/>
    </StaticRouter>
  );
```

I was supposed to get also the async chunks on the `webExtractor.collectChunks()` call.

Any help is appreciated.

Thanks

**Comment by jenniredfield** [comment]
I'll be following this, as it would be interesting to see if there's another way of handling this.

However, I will speculate a bit.

**I think** you always get the main scripts because your server always sends the scripts with entry point 'main'?

I'm not sure how loadable works with react-router... where I work what we did was to not use react-router-dom and let the server do the routing.

For example, something like this...

```
export const ssrApp: RequestHandler = (req,res,next) => { 
...
const currentRoute = req.path; // this would be for example 'page-a'

if(currentRoute === 'page-a') { 
 webExtractor = new ChunkExtractor({ statsFile: webStats, entrypoints: ["page-a"] });
 jsx = webExtractor.collectChunks(
    < AsyncPageA />
  );
}

//... do the rest as usual

}
```

Basically for us, we decided to let the routing be done by the server. So the server checks what page/path you are hitting, and then decides what script entry point to use, and what jsx to render (if in page-a, render markup for page-a) etc.

But I guess this is just one way of doing it.

Would be interesting to see how it's used with react-router-dom.
